crates/hir-ty/src/mapping.rs

```diff
@@ -136,7 +136,8 @@ pub fn from_assoc_type_id(id: AssocTypeId) -> TypeAliasId {
 
 pub fn from_placeholder_idx(db: &dyn HirDatabase, idx: PlaceholderIndex) -> TypeOrConstParamId {
     assert_eq!(idx.ui, chalk_ir::UniverseIndex::ROOT);
-    let interned_id = FromId::from_id(Id::from_u32(idx.idx.try_into().unwrap()));
+    // SAFETY: We cannot really encapsulate this unfortunately, so just hope this is sound.
+    let interned_id = FromId::from_id(unsafe { Id::from_u32(idx.idx.try_into().unwrap()) });
```

Comment:
This is really fishy in general I have to say, salsa even seems to allocate its own `PlaceholderIndex`es so we just gotta hope those don't ever leak out of it 😅 

(this was kind of unsound already etiher way)

Hopefully whatever the new trait solver does allows us to get rid of this weird interning here.

Reply:
> salsa even seems to allocate its own `PlaceholderIndex`es

You mean Chalk?

Reply:
Yes

Reply:
In general I'm not sure how unsafe using an incorrect Id is; it should not cause UB because you can safely create an `Id` then use it on a different database.